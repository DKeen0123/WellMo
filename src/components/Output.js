import React, { Component } from 'react';
import StyledOutput from './styledComponents/Output';

class Output extends Component {
  randomnessGenerator = strings => {
    let string = strings[Math.floor(Math.random() * strings.length)];
    return string;
  };

  badGoodDecider = (badFeelings, goodFeelings) => {
    if (this.props.score > 65) {
      return (
        'You are feeling pretty good right now, ' +
        this.randomnessGenerator(goodFeelings)
      );
    } else {
      return (
        'You are not feeling your best right now, ' +
        this.randomnessGenerator(badFeelings)
      );
    }
  };
  render() {
    const badFeelings = [
      'why not take some time to do something that you know helps you get back to your best',
      'accept how you feel. You feel how you feel and that is OK',
      'think back to a happier time'
    ];

    const goodFeelings = [
      'reflect on the things you have been doing well for yourself, so you can use them to bring you up in future',
      'try and make the most of your positive mood. Why not do some exercise?',
      'time to recognise how good you feel and relish it'
    ];
    return (
      <div>
        <StyledOutput>
          <p id="score">{this.props.score}</p>
        </StyledOutput>
        <p>{this.badGoodDecider(badFeelings, goodFeelings)}</p>
      </div>
    );
  }
}

export default Output;
