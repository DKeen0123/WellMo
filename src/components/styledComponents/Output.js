import styled from 'styled-components';

const StyledOutput = styled.h1`
  font-size: 24px;
`;

export default StyledOutput;
