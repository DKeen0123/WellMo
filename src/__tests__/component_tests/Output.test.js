import React from 'react';
import { shallow, mount } from 'enzyme';
import Output from '../../components/Output';

describe('Output', () => {
  const props = { score: '99' };
  let output = shallow(<Output {...props} />);

  it('renders correctly', () => {
    expect(output).toMatchSnapshot();
  });

  it('displays acore when passed it as props', () => {
    expect(output.find('#score').text()).toEqual('99');
  });
});
