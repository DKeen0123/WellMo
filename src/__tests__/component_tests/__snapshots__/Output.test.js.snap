// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Output renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Output
    score="99"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <p
        id="score"
>
        99
</p>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "99",
        "id": "score",
      },
      "ref": null,
      "rendered": "99",
      "type": "p",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p
          id="score"
>
          99
</p>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "99",
          "id": "score",
        },
        "ref": null,
        "rendered": "99",
        "type": "p",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
